### DISCUSSION TITLE
OffsetDateTime with PostgreSQL

### DISCUSSION THREAD

**Discussion by SteelAlloy** [diesel-rs/diesel#3581] [Ideas]
### Versions

- **Rust: 1.66.0**
- **Diesel: 2.0.3**
- **Database: PostgreSQL**
- **Operating System: Linux**

### Feature Flags

- **diesel: time**

## Problem Description

I'm not sure if it's a missing implementation or if it's misleading documentation. [`OffsetDateTime`](https://docs.diesel.rs/master/time/offset_date_time/struct.OffsetDateTime.html) seems to be available only for the mysql backend.

### What are you trying to accomplish?

Using [`OffsetDateTime`](https://docs.diesel.rs/master/time/offset_date_time/struct.OffsetDateTime.html) when serializing / deserializing from [`Timestamp`](https://docs.diesel.rs/2.0.x/diesel/sql_types/struct.Timestamp.html) with the PostgreSQL backend.

When looking at the source code, this implementation is only covered for mysql.

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues)  and the [disussion forum](https://github.com/diesel-rs/diesel/discussions)  for similar possible closed issues.
 
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [ ] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment]
I converted this to a feature request, as that's not a bug in diesel but more a missing feature. I would probably accept a PR to add the relevant impls for the postgres backend as well.

**Comment by jeff-hiner** [comment]
After some discussion in https://github.com/diesel-rs/diesel/pull/3973 I'm not sure it is a good idea to implement this as asked. There's still a bit of work to be done lining up the TZ-aware types, and maybe some clarification that could be done in the documentation. For my case @Ten0 pointed out it's possible to replace `Timestamp` with `Timestamptz` in the `schema.rs` on Postgres, which a clever workaround that definitely isn't immediately obvious from docs.
